src/message/update-firmware/index.tsx

```diff
@@ -3,35 +3,32 @@ import { Form, notification, Spin } from 'antd';
 import { GenericForm } from '../../components/form';
 import { plainToInstance } from 'class-transformer';
 import { useApiUrl, useCustom } from '@refinedev/core';
-import { ChargingStation } from '../../pages/charging-stations/ChargingStation';
 import { UpdateFirmwareRequest, UpdateFirmwareRequestProps } from './model';
 import { validateSync } from 'class-validator';
 import { MessageConfirmation } from '../MessageConfirmation';
 import { BaseRestClient } from '@util/BaseRestClient';
 import { CHARGING_STATION_SEQUENCES_GET_QUERY } from '../../pages/charging-station-sequences/queries';
 import { formatPem, readFileContent } from '../util';
+import { useSelectedChargingStationIds } from '@hooks';
 
 const DIRECTUS_URL = import.meta.env.VITE_DIRECTUS_URL;
 
-export interface UpdateFirmwareProps {
-  station: ChargingStation;
-}
-
-export const UpdateFirmware: React.FC<UpdateFirmwareProps> = ({ station }) => {
+export const UpdateFirmware: React.FC = () => {
   const formRef = useRef();
+  const apiUrl = useApiUrl();
   const [form] = Form.useForm();
   const [loading, setLoading] = useState(false);
+  const stationIds = useSelectedChargingStationIds();
   const [isFormValid, setIsFormValid] = useState(false);
 
-  const apiUrl = useApiUrl();
   const { data: requestIdResponse, isLoading: isRequestIdLoading } = useCustom({
     url: `${apiUrl}`,
     method: 'post',
     config: { headers: { 'Content-Type': 'application/json' } },
     meta: {
       operation: 'ChargingStationSequencesGet',
       gqlQuery: CHARGING_STATION_SEQUENCES_GET_QUERY,
-      variables: { stationId: station.id, type: 'updateFirmware' },
+      variables: { stationId: stationIds, type: 'updateFirmware' },
```

Comment:
same issue here as with remote start where we will need to account for how this should work for multiple charging stations